src/Symfony/Component/DependencyInjection/Loader/Configurator/ServiceConfigurator.php

```diff
@@ -58,6 +59,11 @@ public function __construct(ContainerBuilder $container, array $instanceof, bool
 
     public function __destruct()
     {
+        if ($this->destructed) {
```

Comment:
This trick prevents the destructor to re-add a service when it is been removed just after being added (see tests).